dependencies/functions/meson.build

```diff
@@ -0,0 +1,73 @@
+# nix check functions
+#============================================================================
+
+if (cpp.has_function('lutimes'))
+    config_h.set(
+        'HAVE_LUTIMES', 1,
+        description : 'Define to 1 if you have the `lutimes` function.')
+endif
+
+if (cpp.has_function('lchown'))
+    config_h.set(
+        'HAVE_LCHOWN', 1,
+        description : 'Define to 1 if you have the `lchown` function.')
+endif
+
+if (cpp.has_function('pipe2'))
+    config_h.set(
+        'HAVE_PIPE2', 1,
+        description : 'Define to 1 if you have the `pipe2` function.')
+endif
+
+if (cpp.has_function('posix_fallocate'))
+    config_h.set(
+        'HAVE_POSIX_FALLOCATE', 1,
+        description : 'Define to 1 if you have the `posix_fallocate` function.')
+endif
+
+if (cpp.has_function('setresuid'))
+    config_h.set(
+        'HAVE_SETRESUID', 1,
+        description : 'Define to 1 if you have the `setresuid` function.')
+endif
+
+if (cpp.has_function('setreuid'))
+    config_h.set(
+        'HAVE_SETREUID', 1,
+        description : 'Define to 1 if you have the `setreuid` function.')
+endif
+
+if (cpp.has_function('statvfs'))
+    config_h.set(
+        'HAVE_STATVFS', 1,
+        description : 'Define to 1 if you have the `statvfs` function.')
+endif
+
+if (cpp.has_function('strsignal'))
+    config_h.set(
+        'HAVE_STRSIGNAL', 1,
+        description : 'Define to 1 if you have the `strsignal` function.')
+endif
+
+if (cpp.has_function('sysconf'))
+    config_h.set(
+        'HAVE_SYSCONF', 1,
+        description : 'Define to 1 if you have the `sysconf` function.')
+endif
```

Comment:
The idiomatic way to do this is with a `foreach` loop:
```meson
foreach f : ['sysconf', 'strsignal', 'statvfs']
  if cpp.has_function(f)
    config.h.set('HAVE_@0@'.format(f.to_upper()), description : 'Define to 1 if you have the `@0@` function.'.format(f))
  endif
endforeach
```

Reply:
this seems to be needlessly complicating things for no reason ..

Reply:
a better way would probably be to assign an array and reference it, instead of calling things inline like the example